packages/meditrak-app/app/database/types/User.js

```diff
@@ -7,9 +7,13 @@ import { Object as RealmObject } from 'realm';
 import { AccessPolicy } from '@tupaia/access-policy';
 
 export class User extends RealmObject {
-  constructor(...args) {
-    super(...args);
-    this.accessPolicy = new AccessPolicy(this.accessPolicyData);
+  accessPolicySingleton = null;
+
+  get accessPolicy() {
+    if (!this.accessPolicySingleton) {
+      this.accessPolicySingleton = new AccessPolicy(this.accessPolicyData);
+    }
+    return this.accessPolicySingleton;
```

Comment:
Most of the changes in this PR are already reviewed in https://github.com/beyondessential/tupaia/pull/1274. It just occurred to me that I could have placed reviewed and non-reviewed changes in different commits 🤔

To assist the reviewer, I will point out which parts are new - so this is **1/3**